extendr-api/src/serializer.rs

```diff
@@ -673,6 +674,17 @@ impl ser::Serialize for Doubles {
     }
 }
 
+impl serde::Serialize for Strings {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        use crate::AsTypedSlice;
+        let strings_slice: &[Rstr] = self.robj.as_typed_slice().unwrap();
```

Comment:
given that we're returning a `Result<>` can we `.map_err()` into `S::Error` and then `?` and same here for `Complexes`

Reply:
The unwrap never fails, as we know it must have this underlying type from the invocation. So I don't see a need to map to any specific error.

Reply:
But let me look closelier....